src/parsers/espresso.js

```diff
@@ -25,6 +29,273 @@ function toEspressoFormat(materialOrConfig) {
+}
+
+/**
+ * @summary Creates cell from ibrav and celldm(i) parameters
+ * @param {Object} system - The system parameters from &SYSTEM namelist
+ * @param {Number} system.ibrav - ibrav parameter
+ * @param {Number[]} system.celldm - celldm parameters
+ * @param {Number} [system.a] - a parameter
+ * @param {Number} [system.b] - b parameter
+ * @param {Number} [system.c] - c parameter
+ * @param {Number} [system.cosab] - cosab parameter
+ * @param {Number} [system.cosac] - cosac parameter
+ * @param {Number} [system.cosbc] - cosbc parameter
+ * @returns {Lattice}
+ */
+function ibravToCell(system) {
+    // eslint-disable-next-line no-unused-vars
+    const { ibrav, celldm, a, b, c, cosab, cosac, cosbc } = system;
+    let alat, type, units;
+
+    // celdm(1) = celldm[0]: index begins from 0
+    if (celldm[0] && a) {
+        throw new Error("Both celldm and A are given");
+    } else if (celldm[0]) {
+        // celldm(x) in bohr
+        // eslint-disable-next-line prefer-destructuring
+        alat = celldm[0];
+        units = "bohr";
+    } else if (a) {
+        // a, b, c, cosAB, cosAC, cosBC in Angstrom
+        alat = a;
+        // eslint-disable-next-line no-unused-vars
+        units = "angstrom";
+    } else {
+        throw new Error("Missing celldm(1)");
+    }
+
+    switch (ibrav) {
+        case 1:
+            type = LATTICE_TYPE.CUB;
+            break;
+        case 2:
+            type = LATTICE_TYPE.FCC;
+            break;
+        case 3:
+            type = LATTICE_TYPE.BCC;
+            break;
+        case -3:
+            type = LATTICE_TYPE.BCC;
+            break;
+        case 4:
+            type = LATTICE_TYPE.HEX;
+            break;
+        case 5:
+            type = LATTICE_TYPE.RHL;
+            break;
+        case -5:
+            type = LATTICE_TYPE.RHL;
+            break;
+        case 6:
+            type = LATTICE_TYPE.TET;
+            break;
+        case 7:
+            type = LATTICE_TYPE.TET;
+            break;
+        case 8:
+            type = LATTICE_TYPE.ORC;
+            break;
+        case 9:
+            type = LATTICE_TYPE.ORCC;
+            break;
+        case -9:
+            type = LATTICE_TYPE.ORCC;
+            break;
+        case 10:
+            type = LATTICE_TYPE.ORCF;
+            break;
+        case 11:
+            type = LATTICE_TYPE.ORCI;
+            break;
+        case 12:
+            type = LATTICE_TYPE.MCL; // Monoclinic P, unique axis c
+            break;
+        case -12:
+            type = LATTICE_TYPE.MCL; // Monoclinic P, unique axis b
+            break;
+        case 13:
+            type = LATTICE_TYPE.MCLC; // Monoclinic P, base centered
+            break;
+        case 14:
+            type = LATTICE_TYPE.TRI;
+            break;
+        default:
+            throw new Error(`ibrav = ${system.ibrav} not implemented`);
+    }
+
+    let vectors = []; // TODO: implement
+    vectors = Lattice.vectorsFromType(type, a, b, c, cosab, cosac, cosbc);
```

Comment:
@timurbazhirov Here the function will receive all parameters converted in accordance with QE specs, while inside of it in ` lattice_bravais` math will be as in https://arxiv.org/pdf/1004.2974.pdf